resources/css/bem/chat-join-channel.less

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+.chat-join-channel {
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+
+  background: hsl(var(--hsl-b4));
+
+  &__channel {
+    display: contents;
+    color: hsl(var(--hsl-l3));
+    text-align: left;
+
+    .link-hover({
+      color: hsl(var(--hsl-c1));
+    });
+
+    &--joined {
+      color: hsl(var(--hsl-c1));
+    }
+  }
+
+  &__channels {
+    display: grid;
+    gap: 10px;
+    padding: 10px;
+
+    // checkmark is 14px wide
+    grid-template-columns: minmax(14px, max-content) 200px auto;
```

Comment:
`... auto 1fr;` is better? (at least on mobile)

Reply:
and as subgrid is a thing now, might as well use it so the button can cover the entire area (instead of having random gaps) and have hover effect

Reply:
guess subgrid should be fine if safari supports it now 🤔